### PR TITLE
csclient: expose API errors

### PR DESCRIPTION
**Pull request by rogpeppe**
We masked errors from several API calls and didn't provide a way to
tell if an error was an authorization error.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/238/
Test PASSed.

File: csclient/csclient.go [hunk 1]
```diff
@@ -2,6 +2,12 @@
 // Licensed under the LGPLv3, see LICENCE file for details.
 
 // The csclient package provides access to the charm store API.
+//
+// Errors returned from the API will usually be of type params.Error.
```
Comment (mhilton) [inline csclient/csclient.go]: *params.Error?

**Comment by ericsnowcurrently** [comment]
LGTM

**Comment by mhilton** [comment]
:+1:

File: csclient/csclient.go (hunk 1 above)
Comment (rogpeppe) [inline csclient/csclient.go]: changed the docs.

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmrepo-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/239/
Test PASSed.

**Pull request merged**